src/main/java/com/ecore/roles/client/UsersClient.java

```diff
@@ -17,21 +18,23 @@
 public class UsersClient {
 
     private final RestTemplate restTemplate;
+    private final RetryTemplate retryTemplate;
     private final ClientsConfigurationProperties clientsConfigurationProperties;
 
     public ResponseEntity<User> getUser(UUID id) {
-        return restTemplate.exchange(
-                clientsConfigurationProperties.getUsersApiHost() + "/" + id,
+        return retryTemplate.execute(context -> restTemplate.exchange(
```

Comment:
Even though UUID is “URL safe”, but this code can use URI Variables in order to follow a consistency of always use URL template variables.